packages/story-editor/src/components/library/panes/media/local/hotlink/useInsert.js

```diff
@@ -78,41 +87,84 @@ function useInsert({ link, setLink, setErrorMsg, onClose }) {
-        resource.src = link;
+        };
+
+        let video;
+
+        // We need to gather some metadata for videos, but efficiently.
+        // Thus only loading it once to speed up insertion.
+        if (isVideo) {
+          // preloadVideoMetadata would suffice, except for audio detection
+          // which requires loading more than just metadata.
+          video = await preloadVideo(proxiedUrl);
+          await seekVideo(video);
+
+          resourceLike.width = video.videoWidth;
+          resourceLike.height = video.videoHeight;
 
-        // We still want to auto-generate video posters, even for hotlinked videos.
-        if ('video' === type && hasUploadMediaAction) {
-          const fileName = getPosterName(
-            originalFileName.replace(`.${ext}`, '')
+          resourceLike.length = Math.round(video.duration);
+          resourceLike.lengthFormatted = getVideoLengthDisplay(
+            resourceLike.length
           );
-          const posterFile = await getFirstFrameOfVideo(proxiedUrl);
-          const posterData = await uploadVideoPoster(0, fileName, posterFile);
-          resource.poster = posterData.poster;
-          resource.posterId = posterData.posterId;
+
+          resourceLike.isMuted = !hasVideoGotAudio(video);
+
+          // We want to auto-generate and *upload* posters for hotlinked videos.
+          // While this could be done _after_ insertion, at this point we have
+          // already preloaded the video, so this will be quick.
+          // Also, this avoids adding a history entry for adding the poster,
+          // which would also cause the autoplayed video to be paused again.
+          // However, upload might fail, thus adding this nested try ... catch.
+          if (hasUploadMediaAction) {
+            try {
+              const fileName = getPosterName(
+                originalFileName.replace(`.${ext}`, '')
+              );
+
+              const posterFile = await getImageFromVideo(video);
+              const posterData = await uploadVideoPoster(
+                0,
+                fileName,
+                posterFile
+              );
+
+              resourceLike.poster = posterData.poster;
+              resourceLike.posterId = posterData.posterId;
+            } catch {
+              // No need to catch poster generation errors.
+            }
+          }
         }
 
+        // Passing the potentially proxied URL here just so that
+        // metadata retrieval works as expected (if still needed after the above).
+        // Afterwards, overriding `src` again to ensure we store the original URL.
+        const resource = await getResourceFromUrl(resourceLike);
+        resource.src = link;
+
         insertElement(type, {
           resource,
         });
 
+        setIsInserting(false);
         setErrorMsg(null);
         setLink('');
         onClose();
       } catch (e) {
+        setIsInserting(false);
         setErrorMsg(getErrorMessage());
       }
```

Comment:
```suggestion
        setErrorMsg(null);
        setLink('');
        onClose();
      } catch (e) {
        setErrorMsg(getErrorMessage());
      } finally {
        setIsInserting(false);
      }
```

Might be cleaner to do this.